core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -438,6 +444,7 @@ private[spark] class ExecutorAllocationManager(
     }
 
     override def onTaskStart(taskStart: SparkListenerTaskStart): Unit = {
+      numRunningTasks += 1
```

Comment:
i think we should let numRunningTasks  to synchronized,because schedule thread and listener  thread are two different thread.